security/rbac/service.go

```diff
@@ -51,54 +59,36 @@ type (
 	}
 
 	enforcerService struct {
-		adapter       persist.BatchAdapter
 		enforcer      casbin.IEnforcer
-		rbacapiLookup apiLookup
+		rbacAPILookup APILookup
 	}
 )
 
 // NewEnforcerSvc creates a new EnforcerService.
-func NewEnforcerSvc(debug bool, adapter persist.BatchAdapter) (Service, error) {
-	model, err := loadModel()
-	if err != nil {
-		return nil, fmt.Errorf("error loading model: %w", err)
-	}
-
-	enforcer, err := casbin.NewEnforcer(model, adapter)
-	if err != nil {
-		return nil, fmt.Errorf("error instantiating casbin enforcer: %w", err)
-	}
-
-	if debug {
-		enforcer.EnableLog(true)
-	}
-
-	rbacapiMappings, err := loadAPIMappings()
-	if err != nil {
-		return nil, fmt.Errorf("error loading rbac api mappings: %w", err)
-	}
-
+func NewEnforcerSvc(enforcer casbin.IEnforcer, rbacAPILookup APILookup) Service {
 	return &enforcerService{
 		enforcer:      enforcer,
-		rbacapiLookup: rbacapiMappings,
-	}, nil
+		rbacAPILookup: rbacAPILookup,
+	}
 }
 
-func loadModel() (model.Model, error) {
+// LoadModel loads the rbac model string from the assethelper and parses it into a Casbin model.Model.
+func LoadModel() (model.Model, error) {
 	modelStr, err := assethelper.LoadRBACModel()
 	if err != nil {
 		return nil, fmt.Errorf("error loading rbac model from assethelper: %w", err)
```

Comment:
Here too, please return something user friendly and log the actual error.

Reply:
@Skarlso I prefer to wrap errors so we can more easily debug. Its up to the caller of this method to log. Therefore I'd prefer adding the logf into `server.go#initRBACService` (for which I definitely forgot!). If this specific error occurred it would be more for us to diagnose a problem than a user issue. So I'd rather include as much context as possible?

Reply:
In that case use `errors.Wrap`?

Reply:
`fmt.Errorf("error loading rbac model from assethelper: %w", err)` is wrapping errors? Its the Go std lib way of wrapping errors introduced in 1.13 :)

Reply:
Ah right, I forgot. :D But then in the upper calls we'll need to do some logic in unwrapping?

Reply:
When you do the log on the error it will show the full error string of all errors. So something like `error a: error b: error c` all the way down the error stack). But other benefits are being able to use `errors.Is` and `errors.As` to look for any errors in the wrapped error.

Reply:
Yes, that's why I asked that we would need to inspect the errors if it's something we can recover from. Anyways, fine then. I just don't want anything to creep up to the user. But I followed through that calls so this should be fine.

Reply:
:+1: Let me just make this code a little bit more consistent dude. You're right though, in some circumstances we'd definitely want to recover. I feel like here if something fails to load (like the model) we're buggered though! Recovery could be turning RBAC off, but that would be bad if it was meant to be on!

Reply:
Yes, let's not do that. :) ( turning it off I mean. )

Reply:
Haha! Don't worry!